feat: :sparkles: integrate Flower (#211)
seedcase-project/template-data-package: docs/guide/develop-data-package.qmd

Author: I moved this file and `justfile.qmd` out of the old `template/docs/README.md`

---
title: "Developing your Data Package"
description: "How to create package properties and resources in your Data Package."
order: 2
---

In your new project generated from the `template-data-package`, the
first steps for creating and developing your Data Package are already
set up in `main.py`. For more detailed instructions on using Seedcase
Sprout to organise your Data Package, see the
[guide](https://sprout.seedcase-project.org/docs/guide/) on Sprout's
website. You can read more about the files and folders created by
`main.py` on the
[Outputs](https://sprout.seedcase-project.org/docs/design/interface/outputs)
page of the design documentation.

## Creating package properties

1. Run `main.py` to create the `scripts/package_properties.py` file for

>    the properties of your Data Package.

Reviewer: ```suggestion
   the properties of your Data Package by using the recipe in the [`justfile`](https://just.systems/man/en/):
```

    ``` bash
    just build
    ```

   You can also run `main.py` by clicking the "Run" button in your IDE.

2. Open `scripts/package_properties.py` and fill in all required fields.
   Also fill in any optional fields you find useful. You can always
   update these later. Make sure to save the file.

3. In `main.py`, uncomment the lines referencing the
   `package_properties` and `package_path` variables.

4. Rerun `main.py` to create the `datapackage.json` file for your Data
   Package.

## Creating a new resource

> ### With data to add to the resource

Reviewer: ```suggestion
### If you already have the data
```

While you can create resource properties without data, it is a lot more
challenging. If at all possible, only create a resource properties
object when you have data to use to at least pre-fill in some of the
important fields. In order to use Sprout, the data needs to already be

> in a tidy format. When it is, load the data as a Polars data frame into

Reviewer: ```suggestion
in a tidy format. When it is, load the data as a [Polars](https://pola.rs) DataFrame into
```
